cli/install/install.go

```diff
@@ -424,30 +462,56 @@ func installTt(binDir string, installCtx InstallCtx, distfiles string) error {
 	// The tag format in tt is vX.Y.Z, but the user can use the X.Y.Z format
 	// and this option needs to be supported.
 	if ttVersion != "master" {
-		versionFound := false
-		for _, ver := range versions {
-			if ttVersion == ver.Str || (ttVersion[0:1] != "v" && "v"+ttVersion == ver.Str) {
-				versionFound = true
-				ttVersion = ver.Str
-				break
+		if isVersion(ttVersion) {
+			log.Infof("Searching in versions...")
+			for _, ver := range versions {
+				if ttVersion == ver.Str || (ttVersion[0:1] != "v" && "v"+ttVersion == ver.Str) {
+					versionFound = true
+					ttVersion = ver.Str
+					break
+				}
+			}
+		} else {
+			log.Infof("Searching in commits...")
+
+			if len(ttVersion) < defaultHashLenght {
+				return fmt.Errorf("the hash must contain at least 7 characters")
+			}
+
+			commits, errHashes := getCommits(installCtx.Local, distfiles, "tt")
+
+			if errHashes != nil {
+				return errHashes
+			}
+
+			croppedHash := ttVersion[0:7]
+
+			for _, commit := range commits {
```

Comment:
Is it possible to use `slices.Contains` here instead of loop?

Reply:
fixed